src/multiint.cpp

```diff
@@ -3517,6 +3522,20 @@ TITLECODE WzMultiOptionTitleUI::run()
 						loadMapSettings1();
 					}
 
+					//Reset player slots if it's a smaller map.
+					if (NetPlay.isHost && NetPlay.bComms && bHosted && !isHoverPreview && oldMaxPlayers > game.maxPlayers)
+					{
+						const std::vector<uint8_t> &Humans = NET_getHumanPlayers();
+
+						//This is pretty ugly tbh, but seems like the easiest way to achieve the goal to my tired mind.
+						size_t PlayerInc = 0;
+						for (uint8_t SlotInc = 0; SlotInc < game.maxPlayers && PlayerInc < Humans.size(); ++SlotInc)
+						{
+							changePosition(Humans[PlayerInc], SlotInc);
```

Comment:
You have to avoid the scavenger slot here:
```
				// Skip the scavenger slot
				if (game.mapHasScavengers && SlotInc == scavengerSlot())
				{
					SlotInc++;
				}
```

Reply:
It doesn't seem to be necessary. game.mapHasScavengers is always true, from what I can tell (hacks), and avoiding the previous scavenger slot just creates floating slots.

Reply:
Well, I had strange floating slot problems with randomization because of the scavenger slot, which were avoided with this code. Why wouldn't it be needed here? You need more than 7 player maps and toggle scavengers on and off to see the issues.

Reply:
Ok, I can reproduce floating slots with and without your suggestion. I'll have check what you did in the randomization option because it never leaves floating slots.

Reply:
@topimiettinen I copied what you did for randomization (see latest commit in this PR) and it no longer has any slot issues that I can see.